src/bootstrap/builder.rs

```diff
@@ -1722,10 +1722,6 @@ impl<'a> Builder<'a> {
             cargo.env("UPDATE_EXPECT", "1");
         }
 
-        if !mode.is_tool() {
-            cargo.env("RUSTC_FORCE_UNSTABLE", "1");
-        }
```

Comment:
It certainly makes no sense to remove this without also removing the code that receives this env var

https://github.com/rust-lang/rust/blob/795ade084ac18e2198faeb63a783d2b8e2108c42/src/bootstrap/bin/rustc.rs#L158-L160

https://github.com/rust-lang/rust/blob/795ade084ac18e2198faeb63a783d2b8e2108c42/src/bootstrap/bin/rustdoc.rs#L55-L57

But this is old code, and I think it is very important. For instance it ensures that if people load the libc crate from the sysroot, they get stability errors, even though libc does not contain any `unstable` markers.

It's generally a bad idea to remove code if you cannot explain why it was there. :)

Reply:
However, if removing this env var helps, that should hopefully get us one step closer to solving the puzzle. Just, the env var has to stay.

Reply:
Yeah, removing it indeed was a bad idea. Will fix the problem by updating the condition instead of removing it.

> It's generally a bad idea to remove code if you cannot explain why it was there. :)

Very true, at the first look it didn't make any sense. But the is the error report so.. :smile: